pkg/apis/agones/v1/gameserver_test.go

```diff
@@ -794,34 +1499,66 @@ func TestGameServerIsBeforeReady(t *testing.T) {
 func TestGameServerApplyToPodContainer(t *testing.T) {
 	t.Parallel()
 
-	name := "mycontainer"
-	gs := &GameServer{
-		Spec: GameServerSpec{
-			Container: name,
-			Template: corev1.PodTemplateSpec{
-				Spec: corev1.PodSpec{
-					Containers: []corev1.Container{
-						{Name: name, Image: "foo/mycontainer"},
-						{Name: "notmycontainer", Image: "foo/notmycontainer"},
+	var testCases = []struct {
+		description string
+		gs          *GameServer
+		errExpected string
```

Comment:
Something to consider: Could pull the expected items into their own struct definition. I find it's a bit easier to read when there are > 1 expected items.

Don't have to do this, but thought I'd suggest it and see if it resonated for you.

Reply:
I like this idea - nice way to separate expected values. Applied all your suggestions, thanks for the review.